src/librustdoc/doctest.rs

```diff
@@ -984,7 +983,7 @@ impl Tester for Collector {
     fn add_test(&mut self, test: String, config: LangString, line: usize) {
         let filename = self.get_filename();
         let name = self.generate_name(line, &filename);
-        let crate_name = self.crate_name.to_string();
+        let crate_name = self.crate_name.clone();
```

Comment:
This seems fine, yeah; creating and executing the doctests binary is orders of magnitude more expensive.